src/main/CommandHandler.cpp

```diff
@@ -768,6 +770,107 @@ CommandHandler::scpInfo(std::string const& params, std::string& retStr)
     retStr = root.toStyledString();
 }
 
+Json::Value
+makeQuorumSetNode(stellar::SCPQuorumSet const& value) {
+    Json::Value node;
+    node["threshold"] = value.threshold;
+
+    Json::Value validators;
+    for (auto const& key : value.validators) {
+        validators.append(KeyUtils::toShortString(key));
+    }
+    node["validators"] = std::move(validators);
+
+    Json::Value innerSets;
+    for (auto const& set : value.innerSets) {
+        innerSets.append(makeQuorumSetNode(set));
+    }
+    node["innerSets"] = std::move(innerSets);
+
+    return node;
+}
+
+void
+CommandHandler::showConfig(std::string const& params, std::string& retStr)
```

Comment:
what do you think about moving most of this to Config class as a toJson method?
then this one could be reduced to retStr = mApp.getConfig().toJson() and it would be easier for someone changing Config class to spot that he/she should also modify the toJson method to reflect those changes?

Reply:
I like the idea. Refactored.